[decomp] Add tests for different dtypes to SDPA decomposition

larryliu0820 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #119239

Summary: As titled. Skipping torch.bfloat16 because for some reason the
difference is 0.01.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

larryliu0820 committed:
[decomp] Add tests for different dtypes to SDPA decomposition

Summary: As titled. Skipping torch.bfloat16 because for some reason the
difference is 0.01.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

drisspg commented on test/test_decomp.py:
```diff
@@ -926,26 +929,27 @@ def test_weight_norm_interface(self, device):
-    def test_sdpa(self, device):
+    @ops(_sdpa_op_info, allowed_dtypes=(torch.float16, torch.float32, torch.float64))
+    def test_sdpa(self, device, dtype, op):
+        # skip torch.bfloat16 because difference is 0.01
```
can we keep the test but just change the tolerances if bfloat16?

larryliu0820 committed:
Update on "[decomp] Add tests for different dtypes to SDPA decomposition"


Summary: As titled. Skipping torch.bfloat16 because for some reason the
difference is 0.01.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

larryliu0820 commented:
@pytorchbot merge

larryliu0820 committed:
Update on "[decomp] Add tests for different dtypes to SDPA decomposition"


Summary: As titled. Skipping torch.bfloat16 because for some reason the
difference is 0.01.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

larryliu0820 commented:
@pytorchbot merge

larryliu0820 committed:
Update on "[decomp] Add tests for different dtypes to SDPA decomposition"


Summary: As titled. Skipping torch.bfloat16 because for some reason the
difference is 0.01.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

larryliu0820 commented:
@pytorchbot merge

larryliu0820 committed:
Update on "[decomp] Add tests for different dtypes to SDPA decomposition"


Summary: As titled. Skipping torch.bfloat16 because for some reason the
difference is 0.01.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

larryliu0820 commented:
@pytorchbot merge
